manifests/charts/istio-control/istio-discovery/templates/reader-clusterrole.yaml

```diff
@@ -37,7 +39,7 @@ rules:
     resources: ["serviceimports"]
     verbs: ["get", "list", "watch"]
   - apiGroups: ["apps"]
-    resources: ["replicasets"]
+    resources: ["replicasets", "deployments"]
```

Comment:
Why does the analyzer require more roles? adding more roles requires justification for many companies.

Reply:
The multi-cluster analyze actually does not require deployment, as its pods already exist in the specified in-cluster format. As for configmap, we only need the injector configmap. However, the analyzers rely on the CRD watcher, which necessitates watch permissions; otherwise, the analysis will not function properly.

Reply:
Our Istiod clusterrole is basically a superset of the permissions granted to the reader service account. The reader service account seems to be out-of-date and even lacks some of our API groups. I was thinking adding only the minimal read permissions necessary for analyze to minimize security risks (This will only run in istiod and istioctl), since we cannot using the istiod clusterrole for multi-cluster setup - it's too risky.